// opengb-sdk/src/models/LobbiesSetPlayerConnectedResponse.ts
// This file is auto-generated by the Open Game Backend (https://opengb.dev) build system.
//
// Do not edit this file directly.
//
// Generated at 2024-08-07T00:51:53.230Z

/* tslint:disable */
/* eslint-disable */

/**
 * @export
 * @interface LobbiesSetPlayerConnectedResponse
 */
export interface LobbiesSetPlayerConnectedResponse {}

/**
 * Check if a given object implements the LobbiesSetPlayerConnectedResponse interface.
 */
export function instanceOfLobbiesSetPlayerConnectedResponse(
	_value: object,
): _value is LobbiesSetPlayerConnectedResponse {
	return true;
}

export function LobbiesSetPlayerConnectedResponseFromJSON(
	json: any,
): LobbiesSetPlayerConnectedResponse {
	return LobbiesSetPlayerConnectedResponseFromJSONTyped(json, false);
}

export function LobbiesSetPlayerConnectedResponseFromJSONTyped(
	json: any,
	ignoreDiscriminator: boolean,
): LobbiesSetPlayerConnectedResponse {
	if (json == null) {
		return json;
	}
	return {};
}

export function LobbiesSetPlayerConnectedResponseToJSON(
	value?: LobbiesSetPlayerConnectedResponse | null,
): any {
	if (value == null) {
		return value;
	}
	return {};
}
