// opengb-sdk/src/models/LobbiesListResponseLobbiesInner.ts
/* tslint:disable */
/* eslint-disable */
/**
 * Open Game Backend
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { mapValues } from '../runtime';
/**
 * 
 * @export
 * @interface LobbiesListResponseLobbiesInner
 */
export interface LobbiesListResponseLobbiesInner {
    /**
     * 
     * @type {string}
     * @memberof LobbiesListResponseLobbiesInner
     */
    id: string;
    /**
     * 
     * @type {string}
     * @memberof LobbiesListResponseLobbiesInner
     */
    version: string;
    /**
     * 
     * @type {{ [key: string]: string; }}
     * @memberof LobbiesListResponseLobbiesInner
     */
    tags: { [key: string]: string; };
}

/**
 * Check if a given object implements the LobbiesListResponseLobbiesInner interface.
 */
export function instanceOfLobbiesListResponseLobbiesInner(value: object): value is LobbiesListResponseLobbiesInner {
    if (!('id' in value) || value['id'] === undefined) return false;
    if (!('version' in value) || value['version'] === undefined) return false;
    if (!('tags' in value) || value['tags'] === undefined) return false;
    return true;
}

export function LobbiesListResponseLobbiesInnerFromJSON(json: any): LobbiesListResponseLobbiesInner {
    return LobbiesListResponseLobbiesInnerFromJSONTyped(json, false);
}

export function LobbiesListResponseLobbiesInnerFromJSONTyped(json: any, ignoreDiscriminator: boolean): LobbiesListResponseLobbiesInner {
    if (json == null) {
        return json;
    }
    return {
        
        'id': json['id'],
        'version': json['version'],
        'tags': json['tags'],
    };
}

export function LobbiesListResponseLobbiesInnerToJSON(value?: LobbiesListResponseLobbiesInner | null): any {
    if (value == null) {
        return value;
    }
    return {
        
        'id': value['id'],
        'version': value['version'],
        'tags': value['tags'],
    };
}
