// opengb-sdk/src/models/LobbiesListResponse.ts
/* tslint:disable */
/* eslint-disable */
/**
 * Open Game Backend
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { mapValues } from '../runtime';
import type { LobbiesListResponseLobbiesInner } from './LobbiesListResponseLobbiesInner';
import {
    LobbiesListResponseLobbiesInnerFromJSON,
    LobbiesListResponseLobbiesInnerFromJSONTyped,
    LobbiesListResponseLobbiesInnerToJSON,
} from './LobbiesListResponseLobbiesInner';

/**
 * 
 * @export
 * @interface LobbiesListResponse
 */
export interface LobbiesListResponse {
    /**
     * 
     * @type {Array<LobbiesListResponseLobbiesInner>}
     * @memberof LobbiesListResponse
     */
    lobbies: Array<LobbiesListResponseLobbiesInner>;
}

/**
 * Check if a given object implements the LobbiesListResponse interface.
 */
export function instanceOfLobbiesListResponse(value: object): value is LobbiesListResponse {
    if (!('lobbies' in value) || value['lobbies'] === undefined) return false;
    return true;
}

export function LobbiesListResponseFromJSON(json: any): LobbiesListResponse {
    return LobbiesListResponseFromJSONTyped(json, false);
}

export function LobbiesListResponseFromJSONTyped(json: any, ignoreDiscriminator: boolean): LobbiesListResponse {
    if (json == null) {
        return json;
    }
    return {
        
        'lobbies': ((json['lobbies'] as Array<any>).map(LobbiesListResponseLobbiesInnerFromJSON)),
    };
}

export function LobbiesListResponseToJSON(value?: LobbiesListResponse | null): any {
    if (value == null) {
        return value;
    }
    return {
        
        'lobbies': ((value['lobbies'] as Array<any>).map(LobbiesListResponseLobbiesInnerToJSON)),
    };
}
