// opengb-sdk/src/models/LobbiesSetLobbyReadyRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * Open Game Backend
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { mapValues } from '../runtime';
/**
 * 
 * @export
 * @interface LobbiesSetLobbyReadyRequest
 */
export interface LobbiesSetLobbyReadyRequest {
    /**
     * 
     * @type {string}
     * @memberof LobbiesSetLobbyReadyRequest
     */
    lobbyId?: string;
    /**
     * 
     * @type {string}
     * @memberof LobbiesSetLobbyReadyRequest
     */
    lobbyToken?: string;
}

/**
 * Check if a given object implements the LobbiesSetLobbyReadyRequest interface.
 */
export function instanceOfLobbiesSetLobbyReadyRequest(value: object): value is LobbiesSetLobbyReadyRequest {
    return true;
}

export function LobbiesSetLobbyReadyRequestFromJSON(json: any): LobbiesSetLobbyReadyRequest {
    return LobbiesSetLobbyReadyRequestFromJSONTyped(json, false);
}

export function LobbiesSetLobbyReadyRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): LobbiesSetLobbyReadyRequest {
    if (json == null) {
        return json;
    }
    return {
        
        'lobbyId': json['lobbyId'] == null ? undefined : json['lobbyId'],
        'lobbyToken': json['lobbyToken'] == null ? undefined : json['lobbyToken'],
    };
}

export function LobbiesSetLobbyReadyRequestToJSON(value?: LobbiesSetLobbyReadyRequest | null): any {
    if (value == null) {
        return value;
    }
    return {
        
        'lobbyId': value['lobbyId'],
        'lobbyToken': value['lobbyToken'],
    };
}
