// opengb-sdk/src/models/LobbiesFindRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * Open Game Backend
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { mapValues } from '../runtime';
/**
 * 
 * @export
 * @interface LobbiesFindRequest
 */
export interface LobbiesFindRequest {
    /**
     * 
     * @type {string}
     * @memberof LobbiesFindRequest
     */
    version: string;
    /**
     * 
     * @type {Array<string>}
     * @memberof LobbiesFindRequest
     */
    regions?: Array<string>;
    /**
     * 
     * @type {{ [key: string]: string; }}
     * @memberof LobbiesFindRequest
     */
    tags?: { [key: string]: string; };
    /**
     * 
     * @type {Array<object>}
     * @memberof LobbiesFindRequest
     */
    players: Array<object>;
    /**
     * 
     * @type {boolean}
     * @memberof LobbiesFindRequest
     */
    noWait?: boolean;
}

/**
 * Check if a given object implements the LobbiesFindRequest interface.
 */
export function instanceOfLobbiesFindRequest(value: object): value is LobbiesFindRequest {
    if (!('version' in value) || value['version'] === undefined) return false;
    if (!('players' in value) || value['players'] === undefined) return false;
    return true;
}

export function LobbiesFindRequestFromJSON(json: any): LobbiesFindRequest {
    return LobbiesFindRequestFromJSONTyped(json, false);
}

export function LobbiesFindRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): LobbiesFindRequest {
    if (json == null) {
        return json;
    }
    return {
        
        'version': json['version'],
        'regions': json['regions'] == null ? undefined : json['regions'],
        'tags': json['tags'] == null ? undefined : json['tags'],
        'players': json['players'],
        'noWait': json['noWait'] == null ? undefined : json['noWait'],
    };
}

export function LobbiesFindRequestToJSON(value?: LobbiesFindRequest | null): any {
    if (value == null) {
        return value;
    }
    return {
        
        'version': value['version'],
        'regions': value['regions'],
        'tags': value['tags'],
        'players': value['players'],
        'noWait': value['noWait'],
    };
}
