// opengb-sdk/src/models/LobbiesCreateResponsePlayersInner.ts
/* tslint:disable */
/* eslint-disable */
/**
 * Open Game Backend
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { mapValues } from '../runtime';
/**
 * 
 * @export
 * @interface LobbiesCreateResponsePlayersInner
 */
export interface LobbiesCreateResponsePlayersInner {
    /**
     * 
     * @type {string}
     * @memberof LobbiesCreateResponsePlayersInner
     */
    token: string;
    /**
     * 
     * @type {string}
     * @memberof LobbiesCreateResponsePlayersInner
     */
    id: string;
}

/**
 * Check if a given object implements the LobbiesCreateResponsePlayersInner interface.
 */
export function instanceOfLobbiesCreateResponsePlayersInner(value: object): value is LobbiesCreateResponsePlayersInner {
    if (!('token' in value) || value['token'] === undefined) return false;
    if (!('id' in value) || value['id'] === undefined) return false;
    return true;
}

export function LobbiesCreateResponsePlayersInnerFromJSON(json: any): LobbiesCreateResponsePlayersInner {
    return LobbiesCreateResponsePlayersInnerFromJSONTyped(json, false);
}

export function LobbiesCreateResponsePlayersInnerFromJSONTyped(json: any, ignoreDiscriminator: boolean): LobbiesCreateResponsePlayersInner {
    if (json == null) {
        return json;
    }
    return {
        
        'token': json['token'],
        'id': json['id'],
    };
}

export function LobbiesCreateResponsePlayersInnerToJSON(value?: LobbiesCreateResponsePlayersInner | null): any {
    if (value == null) {
        return value;
    }
    return {
        
        'token': value['token'],
        'id': value['id'],
    };
}
