// opengb-sdk/src/models/LobbiesFindOrCreateRequest.ts
/* tslint:disable */
/* eslint-disable */
/**
 * Open Game Backend
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { mapValues } from '../runtime';
import type { LobbiesFindOrCreateRequestCreateConfig } from './LobbiesFindOrCreateRequestCreateConfig';
import {
    LobbiesFindOrCreateRequestCreateConfigFromJSON,
    LobbiesFindOrCreateRequestCreateConfigFromJSONTyped,
    LobbiesFindOrCreateRequestCreateConfigToJSON,
} from './LobbiesFindOrCreateRequestCreateConfig';

/**
 * 
 * @export
 * @interface LobbiesFindOrCreateRequest
 */
export interface LobbiesFindOrCreateRequest {
    /**
     * 
     * @type {string}
     * @memberof LobbiesFindOrCreateRequest
     */
    version: string;
    /**
     * 
     * @type {Array<string>}
     * @memberof LobbiesFindOrCreateRequest
     */
    regions?: Array<string>;
    /**
     * 
     * @type {{ [key: string]: string; }}
     * @memberof LobbiesFindOrCreateRequest
     */
    tags?: { [key: string]: string; };
    /**
     * 
     * @type {Array<object>}
     * @memberof LobbiesFindOrCreateRequest
     */
    players: Array<object>;
    /**
     * 
     * @type {boolean}
     * @memberof LobbiesFindOrCreateRequest
     */
    noWait?: boolean;
    /**
     * 
     * @type {LobbiesFindOrCreateRequestCreateConfig}
     * @memberof LobbiesFindOrCreateRequest
     */
    createConfig: LobbiesFindOrCreateRequestCreateConfig;
}

/**
 * Check if a given object implements the LobbiesFindOrCreateRequest interface.
 */
export function instanceOfLobbiesFindOrCreateRequest(value: object): value is LobbiesFindOrCreateRequest {
    if (!('version' in value) || value['version'] === undefined) return false;
    if (!('players' in value) || value['players'] === undefined) return false;
    if (!('createConfig' in value) || value['createConfig'] === undefined) return false;
    return true;
}

export function LobbiesFindOrCreateRequestFromJSON(json: any): LobbiesFindOrCreateRequest {
    return LobbiesFindOrCreateRequestFromJSONTyped(json, false);
}

export function LobbiesFindOrCreateRequestFromJSONTyped(json: any, ignoreDiscriminator: boolean): LobbiesFindOrCreateRequest {
    if (json == null) {
        return json;
    }
    return {
        
        'version': json['version'],
        'regions': json['regions'] == null ? undefined : json['regions'],
        'tags': json['tags'] == null ? undefined : json['tags'],
        'players': json['players'],
        'noWait': json['noWait'] == null ? undefined : json['noWait'],
        'createConfig': LobbiesFindOrCreateRequestCreateConfigFromJSON(json['createConfig']),
    };
}

export function LobbiesFindOrCreateRequestToJSON(value?: LobbiesFindOrCreateRequest | null): any {
    if (value == null) {
        return value;
    }
    return {
        
        'version': value['version'],
        'regions': value['regions'],
        'tags': value['tags'],
        'players': value['players'],
        'noWait': value['noWait'],
        'createConfig': LobbiesFindOrCreateRequestCreateConfigToJSON(value['createConfig']),
    };
}
