// opengb-sdk/src/apis/modules/rivet.ts
// This file is auto-generated by the Open Game Backend (https://opengb.dev) build system.
//
// Do not edit this file directly.
//
// Generated at 2024-08-07T00:51:53.313Z

import * as runtime from "../../runtime";

import type { RivetCallRequest, RivetCallResponse } from "../../models/index";
import {
	RivetCallRequestToJSON,
	RivetCallResponseFromJSON,
} from "../../models/index";

export class Rivet extends runtime.BaseAPI {
	public async call(
		request: RivetCallRequest,
		initOverrides?: RequestInit | runtime.InitOverrideFunction,
	): Promise<RivetCallResponse> {
		const queryParameters: any = {};

		const headerParameters: runtime.HTTPHeaders = {};
		headerParameters["Content-Type"] = "application/json";

		const response = await this.request({
			path: `/modules/rivet/scripts/call/call`,
			method: "POST",
			headers: headerParameters,
			query: queryParameters,
			body: RivetCallRequestToJSON(request),
		}, initOverrides);

		return RivetCallResponseFromJSON(await response.json());
	}
}
